lib/rspec/matchers/built_in/compound.rb

```diff
@@ -36,6 +36,28 @@ def expects_call_stack_jump?
           NestedEvaluator.matcher_expects_call_stack_jump?(matcher_2)
         end
 
+        # @api private
+        # @return [Boolean]
+        def diffable?
+          matcher_1.diffable? || matcher_2.diffable?
+        end
+
+        # @api private
+        # @return [RSpec::Matchers::ExpectedsForMultipleDiffs]
+        def expected
+          return nil unless evaluator
+          ::RSpec::Matchers::ExpectedsForMultipleDiffs.for_many_matchers(matcher_list)
+        end
+
+      protected
+
+        def matcher_list
```

Comment:
I think `matcher_list` is too generic of a name -- outside of the context of diffing/`expected`, I'd expect `compound_matcher.matcher_list` to always return all matchers, not just diffable ones.  WDYT of `matcher_list_for_expected` to make the purpose more explicit?